### PR TITLE
Add publish details on git tags and rake publish command

### PR DESCRIPTION
**Pull request by kylelonghurst**

### CODE REVIEW

File: PUBLISHING.md [hunk 1]
```diff
@@ -4,11 +4,19 @@
 
 This document outlines the process for publishing the Smartsheet for Salesforce User Guide onto Github Pages. The actual Smartsheet for Salesforce documentation can be viewed [here](http://smartsheet-platform.github.io/sfdc-docs). 
 
+## Why git version tags?
+Git tags are a nice way to keep track of the versions of the documentation. The version of the documentation should match the version of the Smartsheet for Salesforce code that it supports.
```
Comment (emilykoh) [inline PUBLISHING.md]: maybe add an example of how to git tag and explanation on how to use semantic versioning

ex. `git tag v1.0.1 after a patch`

Comment (kfansler) [reply PUBLISHING.md]: Yes, please add this.

File: PUBLISHING.md [hunk 2]
```diff
@@ -4,11 +4,19 @@
...
+
+## Understanding the publish command
+The `rake publish` command bundles all of the resources in the `source/` directory to create static assets in a local directory called `bundle/`. These static assesst are then pushed into the `gh-pages` branch. Github pages hosts the contents of the `gh-pages` for public viewing. Only Admins and Owners of the repository have push access into the `gh-pages` repo, so `rake publish` command will fail for anyone else.
```
Comment (kfansler) [inline PUBLISHING.md]: Just a typo: assesst should be assets (I think).

**Review by kfansler — CHANGES_REQUESTED** [review]
I like Emily's example. Please incorporate it. 
Also, there's a typo I flagged.

**Review by kfansler — APPROVED** [review]

**Pull request merged**